File: 73.set-matrix-zeroes/solution.kt
class Solution {
    fun setZeroes(matrix: Array<IntArray>): Unit {
        val m = matrix.size
        val n = matrix[0].size
        val rows = BooleanArray(m)
        val cols = BooleanArray(n)
        for (i in 0 until m) {
            for (j in 0 until n) {
                if (matrix[i][j] == 0) {
                    rows[i] = true
                    cols[j] = true
                }
            }
        }
        for (i in 0 until m) {
            for (j in 0 until n) {
                if (rows[i] || cols[j]) {
                    matrix[i][j] = 0
                }
            }
        }
    }
}

File: 73.set-matrix-zeroes/solution.go
func setZeroes(matrix [][]int)  {
    m, n := len(matrix), len(matrix[0])
    row := make([]bool, m)
    col := make([]bool, n)
    for i := range m {
        for j := range n {
            if matrix[i][j] == 0 {
                row[i] = true
                col[j] = true
            }
        }
    }
    for i := range m {
        for j := range n {
            if row[i] || col[j] {
                matrix[i][j] = 0
            }
        }
    }
}